drivers/hmis/app/graphql/mutations/publish_form_definition.rb

```diff
@@ -25,13 +25,81 @@ def resolve(id:)
       definition.status = Hmis::Form::Definition::PUBLISHED
 
       Hmis::Form::Definition.transaction do
+        # Retire the previously published version
         previous_published_form&.update!(status: Hmis::Form::Definition::RETIRED)
+
+        # Add any missing custom field keys to the form definition (for new questions)
+        new_cdeds = add_missing_custom_field_keys(definition)
+        # Save any new CustomDataElementDefinitions
+        new_cdeds.each(&:save!)
+        # TODO: call new validation function, which should also validate that all referenced CDEDs exist
```

Comment:
So there are attributes that are assigned to the CDED when the field is first published but it looks like they would not be updated on future publications: repeats, field_type, and label.

Maybe this is part of the TODO around validation but should we handle changes to the attributes on existing CDEDs on publish? Maybe label update would be harmless probably but attempts to change the `field_type` and `repeats` attrs could fail a validation check